### PR TITLE
Ports+LibVT+Fonts: Add Ncdu

### PR DESCRIPTION
**Pull request by GMTA**
Add the [`ncdu`](https://dev.yorhel.nl/ncdu) port!

* Modernize ncurses port
* LibVT gets two stubs for keypad control (DECPNM, DECPAM)
* LibVT now supports Cursor Keys Mode (DECCKM)
* LibVT now supports the G0 through G3 working sets
* LibVT now supports VT100 box drawing characters translation
* ~~Added box drawing characters to the Csilla font so we can actually see them~~

image

There are some obvious outstanding issues that can be fixed in future PRs:
* `ncdu` seems to set the foreground color to black (0), and the foreground is only visible because the bold (brightened) variant is used. Not sure why `ncdu` does not set the right (white) color.
* Setting the terminal transparency to something other than 0% results in wonky background colors.
* Resizing the terminal window with `ncdu` open will probably crash Terminal
* When using `--color=dark`, the background color runs do not extend all the way to the right side of the terminal

### CODE REVIEW

File: Userland/Libraries/LibVT/CharacterSet.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+enum CharacterSet {
+    Iso_8859_1,
+    Null,
+    UserDefined,
+    VT100,
+};
+
+class CharacterSetTranslator {
+public:
+    u32 translate_code_point(CharacterSet active_set, u32 code_point)
```
Comment (nico) [inline Userland/Libraries/LibVT/CharacterSet.h:20]: Maybe `translate_vt100_alternate_character_codepoint_to_unicode()`?

File: Userland/Libraries/LibVT/Terminal.h [hunk 2]
```diff
@@ -443,6 +444,10 @@ class Terminal : public EscapeSequenceExecutor {
     Optional<u16> m_column_before_carriage_return;
     bool m_controls_are_logically_generated { false };
     CursorKeysMode m_cursor_keys_mode { Cursor };
+
+    CharacterSetTranslator m_character_set_translator {};
+    size_t m_active_working_set_index { 0 };
```
Comment (nico) [inline Userland/Libraries/LibVT/Terminal.h:449]: Does anything ever modify this?

File: Userland/Libraries/LibVT/Terminal.cpp [hunk 3]
```diff
@@ -1089,7 +1119,38 @@ void Terminal::execute_escape_sequence(Intermediates intermediates, bool ignore,
...
+        case 'K':
+            new_set = CharacterSet::UserDefined;
+            break;
+        default:
+            unimplemented_escape_sequence(intermediates, last_byte);
+            return;
+        }
+
+        dbgln_if(TERMINAL_DEBUG, "Setting G{} working set to character set {}", working_set_index, to_underlying(new_set));
+        m_working_sets[working_set_index] = new_set;
```
Comment (nico) [inline Userland/Libraries/LibVT/Terminal.cpp:1150]: Should this bounds-check `working_set_index`?

File: Userland/Libraries/LibVT/CharacterSet.h (hunk 1 above)
Comment (GMTA) [reply Userland/Libraries/LibVT/CharacterSet.h:20]: Not really - it should also support three other sets apart from VT100, so I could extract a method for just VT100 but I'd rather see a nicer implementation of translation tables.

File: Userland/Libraries/LibVT/Terminal.h (hunk 2 above)
Comment (GMTA) [reply Userland/Libraries/LibVT/Terminal.h:449]: Not yet - there are some as of yet unimplemented sequences like Ctrl+O or Ctrl+N that should write to this variable.

File: Userland/Libraries/LibVT/Terminal.cpp (hunk 3 above)
Comment (nico) [reply Userland/Libraries/LibVT/Terminal.cpp:1150]: Ah, I see. Maybe add a `VERIFY()` to make this more obvious, but I suppose it's pretty obvious already :)

**Comment by nico** [comment]
Looks good to me.

File: Userland/Libraries/LibVT/Terminal.cpp (hunk 3 above)
Comment (GMTA) [reply Userland/Libraries/LibVT/Terminal.cpp:1151]: `VERIFY()` is now in!

**Comment by linusg** [comment]
Font conflicts

**Comment by GMTA** [comment]
> Font conflicts

@linusg you merged the newer #10306 before this one, so that's a conflict alright. On Discord someone drew all the other box drawing characters so I'm not sure I should invest more time in this. I'll probably leave out the font update altogether.

**Comment by linusg** [comment]
Sorry about that! PRs rarely get merged in creation order, I get that it's annoying though.

> On Discord someone drew all the other box drawing characters so I'm not sure I should invest more time in this. I'll probably leave out the font update altogether.

Sure, that works for me :)

**Comment by GMTA** [comment]
@linusg Font changes are now removed from this PR!

**Review by linusg — APPROVED** [review]

**Pull request merged**